pkg/virt-controller/watch/migration.go

```diff
@@ -599,15 +587,16 @@ func (c *MigrationController) processMigrationPhase(
 	case virtv1.MigrationRunning:
 		_, exists := pod.Annotations[virtv1.MigrationTargetReadyTimestamp]
 		if !exists && vmi.Status.MigrationState.TargetNodeDomainReadyTimestamp != nil {
-			key := patch.EscapeJSONPointer(virtv1.MigrationTargetReadyTimestamp)
-			patchOps := fmt.Sprintf(`[{ "op": "add", "path": "/metadata/annotations/%s", "value": "%s" }]`,
-				key,
-				vmi.Status.MigrationState.TargetNodeDomainReadyTimestamp.String())
-
-			_, err := c.clientset.CoreV1().Pods(pod.Namespace).Patch(context.Background(), pod.Name, types.JSONPatchType, []byte(patchOps), v1.PatchOptions{})
+			patchBytes, err := patch.New(
+				patch.WithAdd(fmt.Sprintf("/metadata/annotations/%s", patch.EscapeJSONPointer(virtv1.MigrationTargetReadyTimestamp)), vmi.Status.MigrationState.TargetNodeDomainReadyTimestamp.String()),
```

Comment:
IIRC, @iholder101 wanted to include the EscapeJSONPointer directly in the methods in the patch package. @fossedihelm do you mind to add this there? It will make the code cleaner.
Like in `pkg/apimachinery/patch/patch.go`:
```golang 
func (p *PatchSet) addOp(op, path string, value interface{}) {
        pathNoPointer := EscapeJSONPointer(path)
        p.patches = append(p.patches, PatchOperation{
                Op:    op,
                Path:  pathNoPointer,
                Value: value,
        })
}
```
In this way, when we have replaced all the patch operations we can set that method as private

Reply:
If you don't want to include this here, I can open a separate pr, let me know

Reply:
I intend to include `EscapeJSONPointer` directly there.
But I want to do it in an extreme way:
I want to unexport it and avoid all the external usages(without `patchSet`) of it first.
This is because if you accidentally use `EscapeJSONPointer` twice it will result in a wrong format. Example:
- t0: `kubevirt.io/something~`
- t1: first `EscapeJSONPointer` ->  `kubevirt.io~1something~0`
- t2: second `EscapeJSONPointer` -> `kubevirt.io~01something~00`
How about addressing it in a separate PR once this is merged?

Reply:
ok, make sense